base/range.jl

```diff
@@ -426,26 +427,30 @@ value `r[1]`, but alternatively you can supply it as the value of
 with `TwicePrecision` this can be used to implement ranges that are
 free of roundoff error.
 """
-struct StepRangeLen{T,R,S} <: AbstractRange{T}
+struct StepRangeLen{T,R,S,L} <: AbstractRange{T}
```

Comment:
```suggestion
struct StepRangeLen{T,R,S,L<:Integer} <: AbstractRange{T}
```
Or is there a reason to leave `L` unconstrained?

Reply:
👍 it is limited by the (only) constructor, so probably makes sense to limit here too